lib/slack-ruby-bot/app.rb

```diff
@@ -4,7 +4,9 @@ module SlackRubyBot
   class App < Server
     def initialize(options = {})
       SlackRubyBot.configure do |config|
```

Comment:
So I think this old code is wrong. We're trying to set a global config token in the constructor for a server. Let's move this code into a class method and rewrite `self.instance`.

```ruby
def self.instance
   @instance ||= begin
        configure!
        new
   end
end
```

Constructor should also take an optional token, this way we can have many server instances with different tokens. It's not used like this, but it makes testing easier and doesn't mix instances and globals?